pyclam/manifold.py

```diff
@@ -538,95 +659,227 @@ def subgraph(self, cluster: 'Cluster') -> 'Graph':  # TODO: Cover
 
     def clear_cache(self) -> None:
         """ Clears the cache of the graph. """
+        # Clear all cached values and edges.
         self.cache = {'optimal': self.cache['optimal']}
-        # Clear all cached edges.
-        self.clusters = {c: None for c in self.clusters.keys()}
+        self.edges = {cluster: None for cluster in self.edges}
         return
 
-    def neighbors(self, cluster: Cluster) -> Dict[Cluster, float]:
-        """ return all neighbors of a given cluster. """
-        if self.clusters[cluster] is None:
+    # noinspection DuplicatedCode
+    def neighbors(
+            self,
+            cluster: Cluster,
+            *,
+            choice: str = 'all',
+    ) -> List[Cluster]:
+        """ return neighbors of a given cluster.
+
+        :param cluster: source cluster
+        :param choice: 'all' for all neighbors,
+                       'transition' for only those that are not subsumed,
+                       'subsumed' for only those that are subsumed.
+        :return:
+        """
+        if self.edges[cluster] is None:
             self.build_edges()
-        return {edge.neighbor: edge.distance for edge in self.clusters[cluster]}
 
-    def distances(self, cluster: Cluster) -> List[float]:
+        if choice == 'all':
+            return [edge.neighbor for edge in self.edges[cluster]]
+        elif choice == 'transition':
+            return [edge.neighbor for edge in self.transition_edges[cluster]]
+        elif choice == 'subsumed':
+            return [edge.neighbor for edge in self.subsumed_edges[cluster]]
+        else:
+            raise ValueError(f'choice must be one of: '
+                             f'\'all\', '
+                             f'\'transition\', or '
+                             f'\'subsumed\'. '
+                             f'Got: {choice}')
+
+    # noinspection DuplicatedCode
+    def distances(
+            self,
+            cluster: Cluster,
+            *,
+            choice: str = 'all',
```

Comment:
should always put all of the options in the docstring imo, I personally dislike having to pass a string around too fwiw, hate losing autocomplete